### PR TITLE
KOGITO-535: securing the GraphQL endpoint using quarkus security conf

### PR DESCRIPTION
**Pull request by nmirasch**
Add quarkus security configuration to manage data-index-service security
Disable graphiql UI when open id auth is enabled

### CODE REVIEW

File: data-index/data-index-service/src/main/resources/application.properties [hunk 1]
```diff
@@ -9,7 +9,6 @@ quarkus.log.category."graphql".level=INFO
 # Quarkus HTTP
 quarkus.http.port=8180
 quarkus.http.test-port=8181
-quarkus.http.cors=true
```
Comment (cristianonicolai) [inline data-index/data-index-service/src/main/resources/application.properties]: shouldnt cors be enable indepently from auth?

Comment (nmirasch) [reply data-index/data-index-service/src/main/resources/application.properties]: @cristianonicolai I can leave it but regarding it is to allow to share restricted resources, from another domain  I thought without auth we don't have it. But it doesn't hurt, I can leave it as it was

Comment (mswiderski) [reply data-index/data-index-service/src/main/resources/application.properties]: I guess it won't work in travel agency app as soon cors is not enabled...

Comment (cristianonicolai) [reply data-index/data-index-service/src/main/resources/application.properties]: yeah, that was my concern.

**Comment by nmirasch** [comment]
@mswiderski @cristianonicolai rebased. About the enabling of cors, I keep it enabled even the auth is disabled.

**Comment by mswiderski** [comment]
@nmirasch would you mind to have a look why the build failed?

**Comment by nmirasch** [comment]
Jenkins retest this please

**Comment by nmirasch** [comment]
Jenkins retest this please

**Comment by nmirasch** [comment]
@cristianonicolai  @mswiderski I have moved this pr to another branch in this PR https://github.com/kiegroup/kogito-runtimes/pull/236
I'm going to close this one and ask for  your  review there. Thanks!

**Pull request closed without merging**